phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -0,0 +1,459 @@
+                            }
+                            LOGGER.info("Unloaded IndexRegionObserver coprocessor on data table "+fullTableName);
+                        }
+                        if (!admin.getDescriptor(TableName.valueOf(fullTableName)).hasCoprocessor(Indexer.class.getName())) {
+                            if (!dryRun) {
+                                dataTableDescBuilder.addCoprocessor(Indexer.class.getName(), null, QueryServicesOptions.DEFAULT_COPROCESSOR_PRIORITY, prop);
+                            }
+                            LOGGER.info("Loaded Indexer coprocessor on data table "+fullTableName);
+                        }
+                        if (!dryRun) {
+                            admin.modifyTable(dataTableDescBuilder.build());
+                        }
+                    }
+                    for (String indexName : entry.getValue()) {
+                        if (!dryRun) {
+                            admin.disableTable(TableName.valueOf(indexName));
+                        }
+                        LOGGER.info("Disabled index table "+ indexName);
+                        TableDescriptorBuilder indexTableDescBuilder = TableDescriptorBuilder.newBuilder(admin.getDescriptor(TableName.valueOf(indexName)));
+                        if (admin.getDescriptor(TableName.valueOf(indexName)).hasCoprocessor(GlobalIndexChecker.class.getName())) {
+                            indexTableDescBuilder.removeCoprocessor(GlobalIndexChecker.class.getName());
+                            LOGGER.info("Unloaded GlobalIndexChecker coprocessor on index table "+indexName);
+                        }
+                        if (!dryRun) {
+                            admin.modifyTable(indexTableDescBuilder.build());
+                        }
+                    }
+                    for (String indexName : entry.getValue()) {
+                        if (!dryRun) {
+                            admin.enableTable(TableName.valueOf(indexName));
+                        }
+                        LOGGER.info("Enabled index table "+ indexName);
+                    }
+                    if (!dryRun) {
+                        admin.enableTable(TableName.valueOf(fullTableName));
+                    }
+                    LOGGER.info("Enabled data table "+ fullTableName);
+                } catch (IOException | SQLException e) {
+                    LOGGER.severe("Something went wrong while rolling back the feature "+ e);
+                }
+            }
+
+        } catch (SQLException e) {
+            LOGGER.severe("Something went wrong in rollbackConsistencyFeature "+ e);
+        }
+    }
+
+    private static void checkAndWaitForMajorCompaction(Admin admin, Map.Entry<String, ArrayList<String>> value) {
+        value.getValue().parallelStream().forEach(table -> {
+            try {
+                CompactionState cs = admin.getCompactionState(TableName.valueOf(table));
```

Comment:
Does this return MAJOR when any of the table regions is running Major compaction? What happens when some are running Minor compaction and some are running Major compaction?

Reply:
status is MAJOR_AND_MINOR. 
updated the code.